### PR TITLE
Pipeline refactor and upgrade pipelines (NR-95398)

### PR DESCRIPTION
**Pull request by kang-makes**

### CODE REVIEW

File: .github/workflows/push_pr.yml [hunk 1]
```diff
@@ -103,16 +91,60 @@ jobs:
...
+        with:
+          name: windows-packages
+          path: src/github.com/${{ env.ORIGINAL_REPO_NAME }}/dist/
+
+      - name: Get PFX certificate from GH secrets
+        shell: bash
+        run: printf "%s" "$PFX_CERTIFICATE_BASE64" | base64 -d - > wincert.pfx
+      - name: Extract .exe
+        shell: pwsh
+        run: build\windows\extract_exe.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG"
```
Comment (sigilioso) [inline .github/workflows/push_pr.yml:147]:
```suggestion
        run: .\build\windows\extract_exe.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG"
```
is `.\` needed? I'm not sure...

On the other hand, it seems `build/windows/download_zip_extract_exe.ps1` was split, but the corresponding script where not included. Maybe that is the reason why the [pipeline is failing at the moment](https://github.com/newrelic/nri-mssql/actions/runs/4424333599/jobs/7758075973).

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -142,11 +122,11 @@ jobs:
       - name: Download zip from GH Release assets and extract .exe
         shell: pwsh
         run: |
-          build\windows\download_zip_extract_exe.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG" "$env:REPO_FULL_NAME"
+          build\windows\download_zip.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG" "$env:REPO_FULL_NAME"
+          build\windows\extract_exe.ps1 "$env:INTEGRATION" ${{ matrix.goarch }} "$env:TAG"
```
Comment (sigilioso) [inline .github/workflows/prerelease.yml:126]: The same that the previous comment in the `push_pr` workflow.

**Review by sigilioso — COMMENTED** [review]
Thanks for taking care of it. I left a couple of comments (both are merely the same)

File: .github/workflows/prerelease.yml (hunk 2 above)
Comment (kang-makes) [reply .github/workflows/prerelease.yml:126]: Let's keep the conversation in the [same thread](https://github.com/newrelic/nri-mssql/pull/108#discussion_r1136815893). I am marking this as resolved.

File: .github/workflows/push_pr.yml (hunk 1 above)
Comment (kang-makes) [reply .github/workflows/push_pr.yml:147]: `.\` is not needed in Windows.

And you are right, I did a `git commit -am ""` without adding new files created after splitting the previous one.

Commit [#ec2fffe](https://github.com/newrelic/nri-mssql/pull/108/commits/ec2fffe1dc9e2b1082c4f5408c48b1f168a3df45) should solve the issue.

**Review by sigilioso — COMMENTED** [review]
Would it make sense to include the 'are the env vars defined' check (concerning secrets) in this repository as well?

**Comment by kang-makes** [comment]
Hi @sigilioso,

You are right. I left it out because we were still deciding if it is going to make it as the final solution or not.

Let me know if there is anything missing because I think I have finished with this PR :)

**Review by sigilioso — APPROVED** [review]
I think that was it, thanks for taking care of it! 👍

**Pull request merged**